Functions/Example/Tests/FIRFunctionsTests.m

```diff
@@ -81,6 +83,29 @@ - (void)tearDown {
   [super tearDown];
 }
 
+- (void)testFunctionsInstanceIsStablePerApp {
+  FIROptions *options =
+      [[FIROptions alloc] initWithGoogleAppID:@"0:0000000000000:ios:0000000000000000"
+                                  GCMSenderID:@"00000000000000000-00000000000-000000000"];
+  [FIRApp configureWithOptions:options];
+
+  FIRFunctions *functions1 = [FIRFunctions functions];
+  FIRFunctions *functions2 = [FIRFunctions functionsForRegion:@"us-central"];
+
+  XCTAssertEqualObjects(functions1, functions2);
+
+  [FIRApp configureWithName:@"test" options:options];
+  FIRApp *app2 = [FIRApp appNamed:@"test"];
+
+  functions2 = [FIRFunctions functionsForApp:app2 region:@"us-central"];
+
+  XCTAssertNotEqualObjects(functions1, functions2);
+
+  functions1 = [FIRFunctions functionsForApp:app2 customDomain:@"test_domain"];
+
+  XCTAssertEqualObjects(functions1, functions2);
```

Comment:
Providing a custom domain should be a different instance of Functions, no?

Reply:
See above comment for reasoning.